xdem/coreg/affine.py

```diff
@@ -385,27 +390,30 @@ def _apply_pts(
 
         """Apply the VerticalShift function to a set of points."""
         dem_copy = elev.copy()
-        dem_copy[z_name] += self._meta["vshift"]
+        dem_copy[z_name] += self._meta["shift_z"]
         return dem_copy
 
     def _to_matrix_func(self) -> NDArrayf:
         """Convert the vertical shift to a transform matrix."""
         empty_matrix = np.diag(np.ones(4, dtype=float))
 
-        empty_matrix[2, 3] += self._meta["vshift"]
+        empty_matrix[2, 3] += self._meta["shift_z"]
 
         return empty_matrix
 
 
 class ICP(AffineCoreg):
     """
-    Iterative Closest Point DEM coregistration.
-    Based on 3D registration of Besl and McKay (1992), https://doi.org/10.1117/12.57955.
+    Iterative closest point registration, based on Besl and McKay (1992), https://doi.org/10.1117/12.57955.
+
+    Estimates a rigid transform (rotation + translation) between two elevation datasets.
 
-    Estimates a rigid transform (rotation + translation) between two DEMs.
+    The transform is stored in the `self.meta` key "matrix", with rotation centered on the coordinates in the key
+    "centroid". The translation parameters are also stored individually in the keys "shift_x", "shift_y" and "shift_z"
```

Comment:
But these may be misleading, if there is a rotation, no? Should there also be an item "rotation" in the meta?

Reply:
A rotation here is quite complex (3D) and already contained in the affine matrix. Maybe we can have a note pointing at `pytransform3d` that has routines to convert the affine matrix into its rotation/translation/scaling components? (it's really not trivial)
We could also include the components as `rotation_x`, `rotation_y`, `rotation_z`.
Or just do both!